src/features/events/store.ts

```diff
@@ -515,32 +515,50 @@ const eventsSlice = createSlice({
 
 function addEventToState(state: EventsStoreSlice, events: ZetkinEvent[]) {
   events.forEach((event) => {
-    const dateStr = event.start_time.slice(0, 10);
-
-    if (!state.eventsByDate[dateStr]) {
-      state.eventsByDate[dateStr] = remoteList();
+    const newDateStr = event.start_time.slice(0, 10);
+    if (!state.eventsByDate[newDateStr]) {
+      state.eventsByDate[newDateStr] = remoteList();
```

Comment:
I think this bit could move down to where the other things are that concerns `eventsByDate`, so that the code is grouped with first `eventList`, then `eventsByDate`, and last `eventsByCampaignId`